cognite/gqlpygen/main.py

```diff
@@ -1,60 +1,165 @@
-    client_name = to_client_name(name)
-    schema_name = to_schema_name(name)
+    client_name = to_client_name(prefix)
+    schema_name = to_schema_name(prefix)
     sdk = to_client_sdk(schema_raw, client_name, schema_name)
+    output_dir = (output_dir or graphql_schema.parent).absolute()
+    output_dir.mkdir(exist_ok=True)
 
     for name, content in sdk.items():
-        output = Path.cwd() / name
+        output = output_dir / name
         output.write_text(content)
         click.echo(f"Wrote file {output.relative_to(Path.cwd())}")
 
 
 @app.command("togql", help="Input a pydantic schema to create .graphql schema")
-def to_gql(pydantic_schema: Path = typer.Argument(..., help="Pydantic schema to convert")):
-    module_name = pydantic_schema.stem
-    click.echo(f"Got file {pydantic_schema}, trying to import it...")
-    spec = importlib.util.spec_from_file_location(module_name, pydantic_schema)
-    module = importlib.util.module_from_spec(spec)  # type: ignore [arg-type]
-    sys.modules[module_name] = module
-    spec.loader.exec_module(module)  # type: ignore [union-attr]
+def to_gql(
+    schema_module: Path = typer.Argument(
+        _schema_module or ..., help="Pydantic schema to convert. Path to a .py file or Python dot notation "
+    ),
+    graphql_schema: Path = typer.Option(_graphql_schema or ..., help="File path for the output."),
+    prefix: str = typer.Option(_prefix, help="Name prefix for the domain."),
+):
+    if str(schema_module).endswith(".py"):
+        click.echo(f"Got file {schema_module}, trying to import it...")
+        module_name = schema_module.stem
+        spec = importlib.util.spec_from_file_location(module_name, schema_module)
+        module = importlib.util.module_from_spec(spec)  # type:ignore[arg-type]
+        sys.modules[module_name] = module
+        spec.loader.exec_module(module)  # type:ignore[union-attr]
+    else:
+        module_name = settings.local.get("schema_module", default=schema_module.stem)
+        click.echo(f"Got module {schema_module}, trying to import it...")
+        module = importlib.import_module(module_name)
     click.echo("Import successful")
 
-    click.echo("Searching for a schema...")
-    for schema_name, instance in inspect.getmembers(module):
-        if isinstance(instance, Schema):
-            click.echo(f"Found schema {schema_name!r}")
-            break
+    if not _prefix:
+        click.echo("Searching for a schema...")
+        for schema_name, instance in inspect.getmembers(module):
+            if isinstance(instance, Schema):
+                click.echo(f"Found schema {schema_name!r}")
+                break
+        else:
+            click.echo("Failed to find schema, exiting..")
+            exit(1)
     else:
-        click.echo("Failed to find schema, exiting..")
-        exit(1)
+        schema_name = to_schema_name(prefix)
+        click.echo(f"Got schema name {schema_name}")
+        instance = getattr(module, schema_name)
```

Comment:
This should catch the error if the schema_name is wrong, and give a user friendly message.

Reply:
https://github.com/cognitedata/cognite-gql-pygen/pull/20/commits/b2d5a890f69842058ebbdf0a0d4d1ae6a9c1de63